tests/test_protocol_hardening.py

```
@@ -95,17 +95,16 @@ def test_remove_transactions_by_sender_nonce_when_tx_id_differs(self):
 
 class TestP2PValidationAndDedup(unittest.IsolatedAsyncioTestCase):
     async def test_invalid_message_schema_is_rejected(self):
-        network = P2PNetwork()
-
-        invalid_message = {"type": "tx", "data": {"sender": "abc"}}
-        self.assertFalse(network._validate_message(invalid_message))
+        invalid_payload = {"sender": "abc"}
+        with self.assertRaises(Exception):
```

Ruff (0.15.17)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)